extensions/snowflake/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+enum SnowflakeError {
+    #[error("Failed to connect to Snowflake: {0}")]
+    Connection(String),
+    #[error("Failed to execute query: {0}")]
+    Query(String),
+    #[error("Pool error: {0}")]
+    Pool(#[from] PoolError),
+}
+
+struct SnowflakeManager {
+    config: SnowflakeConfig,
+}
+
+impl Manager for SnowflakeManager {
+    type Type = SnowflakeClient;
+    type Error = SnowflakeError;
+
+    async fn create(&self) -> Result<SnowflakeClient, SnowflakeError> {
+        let client = ClientBuilder::new()
+            .account(&self.config.account)
+            .username(&self.config.username)
+            .password(&self.config.password)
+            .database(&self.config.database)
+            .warehouse(&self.config.warehouse)
+            .role(self.config.role.as_deref().unwrap_or("PUBLIC"))
+            .build()
+            .map_err(|e| SnowflakeError::Connection(e.to_string()))?;
+
+        Ok(client)
+    }
+
+    async fn recycle(&self, _client: &mut SnowflakeClient) -> Result<(), SnowflakeError> {
+        // Could implement connection testing here if needed
+        Ok(())
+    }
+}
+
+#[derive(ResolverExtension)]
+struct SnowflakeExtension {
+    pool: Arc<Pool<SnowflakeManager>>,
+}
+
+#[derive(Deserialize)]
+struct SnowflakeConfig {
+    account: String,
+    username: String,
+    password: String,
+    database: String,
+    warehouse: String,
+    role: Option<String>,
+}
+
+#[derive(Deserialize)]
+struct QueryArgs {
+    sql: String,
+    params: Option<Vec<String>>,
+}
+
+#[derive(Serialize)]
+struct QueryResult {
+    rows: Vec<serde_json::Value>,
+    affected_rows: Option<i64>,
+}
+
+impl Extension for SnowflakeExtension {
+    fn new(schema_directives: Vec<Directive>, config: Configuration) -> Result<Self, Box<dyn std::error::Error>> {
+        let config: SnowflakeConfig = config.deserialize()?;
+        info!("Initializing Snowflake extension with account: {}", config.account);
+
+        let manager = SnowflakeManager { config };
+        let pool = Pool::builder(manager)
+            .max_size(10)
+            .build()
+            .map_err(|e| Box::new(SnowflakeError::Pool(e)))?;
+
+        Ok(Self {
+            pool: Arc::new(pool),
+        })
+    }
+}
+
+impl Resolver for SnowflakeExtension {
+    fn resolve_field(
+        &mut self,
+        _context: SharedContext,
+        directive: Directive,
+        field_definition: FieldDefinition,
+        inputs: FieldInputs,
+    ) -> Result<FieldOutput, Error> {
+        let args: QueryArgs = directive.arguments()?;
+        debug!("Executing Snowflake query: {}", args.sql);
+
+        // Get a connection from the pool
+        let client = self.pool.get().map_err(|e| Error {
+            message: format!("Failed to get connection from pool: {}", e),
+            extensions: vec![],
+        })?;
+
+        // Replace parameter placeholders with actual values
+        let sql = if let Some(params) = args.params {
+            let mut sql = args.sql;
```

Comment:
Are you sure the abstraction for just passing an SQL string is what we want here?